tests/unit/services/test_project.py

```diff
@@ -227,45 +269,100 @@ def test_expand_environment_no_partitions_any_platform(
     ],
 )
 def test_expand_environment_for_riscv64(
-    project_service: ProjectService, project_data, expected, fake_host_architecture
+    real_project_service: ProjectService,
+    project_data,
+    expected,
+    fake_host_architecture,
+    fake_platform,
 ):
-    project_service._expand_environment(project_data, "riscv64")
+    real_project_service._expand_environment(
+        project_data,
+        platform=fake_platform,
+        build_for="riscv64",
+        build_on=fake_host_architecture,
+    )
     assert project_data == expected
 
 
 @pytest.mark.parametrize(
-    "build_for", [arch.value for arch in craft_platforms.DebianArchitecture] + ["all"]
+    "build_for", [arch.value for arch in craft_platforms.DebianArchitecture]
 )
 @pytest.mark.usefixtures("enable_partitions")
 def test_expand_environment_stage_dirs(
```

Comment:
note: I am confused about this test. I failed to understand why it using the platform/build_for values as partition names.

Reply:
That's to test that we have variable/flexible partitions based on the platform. We're using the `FakeProjectService` from `conftest.py` here which does that. I've added a comment to explain.